components/merkledb/src/validation.rs

```diff
@@ -12,26 +23,43 @@ pub fn is_valid_index_name<S: AsRef<str>>(name: S) -> bool {
 /// Check that latin1 character is allowed in index name.
 /// Only these combination of symbols are allowed:
 ///
-/// `[0..9]`, `[a-z]`, `[A-Z]`, `_`, `-`, `.`
+/// `[0..9]`, `[a-z]`, `[A-Z]`, `_`, `-`
 pub fn is_allowed_latin1_char(c: u8) -> bool {
     match c {
-        48..=57   // 0..9
+        48..=57     // 0..9
```

Comment:
Hm. Maybe use byte character literals? `b'0'..=b'9'` is much more self-explanatory.
Same applies for `is_valid_index_name`: `b'.'` is better than 46 IMO.